imbi/models.py

```diff
@@ -457,6 +472,27 @@ async def project_components(
     return await _load_collection(ProjectComponent, project_id, application)
 
 
+async def project_dependency(
+        project_id: int, dependency_id: int,
+        application: 'app.Application') -> ProjectDependency:
+    def on_postgres_error(_metric_name: str, exc: Exception) -> None:
+        LOGGER.error(
+            'Failed to execute query for project dependency '
+            '(project_id: %s, dependency_id: %s): %s', project_id,
+            dependency_id, exc)
+        raise errors.DatabaseError('Error loading ProjectDependency',
+                                   error=exc)
+
+    async with application.postgres_connector(
+            on_error=on_postgres_error) as conn:
+        result = await conn.execute(ProjectDependency.SQL, {
+            'project_id': project_id,
+            'dependency_id': dependency_id
+        }, 'model-load')
+        if result.row_count:
+            return ProjectDependency(**result.row)
+
+
```

Comment:
What do you think about changing the `_load` function to be a little more flexible?
```python
async def _load(model: dataclasses.dataclass, application: 'app.Application',
                **key: object) -> dataclasses.dataclass:
    def on_postgres_error(_metric_name: str, exc: Exception) -> None:
        LOGGER.error('Failed to execute query for %s with key %r: %s',
                     model.__class__.__name__, key, exc)
        raise errors.DatabaseError(f'Error loading {model.__class__.__name__}',
                                   error=exc)

    async with application.postgres_connector(
            on_error=on_postgres_error) as conn:
        result = await conn.execute(model.SQL, key, 'model-load')
        if result.row_count:
            return model(**result.row)
```

Turns out that it is only used three times so updating from `_load(WhateverType, some_id, application)` to `_load(WhateverType, application, id=some_id)` is a pretty small change. Then you can reuse it here.

Reply:
There is also an easter egg in the current implementation... it _always_ reports `Failed to execute query for project {id}: {error}` even though it is only used for namespaces, project types, and operation log entries 😞

Reply:
Good idea, I couldn't quite think of it when I was working on this.